File: docs/algorithms/code/algorithms/levelOrder.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function levelOrder(root: TreeNode | null): number[][] {
  if (!root) {
    return [];
  }

  const res: number[][] = [];

  const stack: TreeNode[] = [root];

  while (stack.length) {
    let size = stack.length;
    const record: number[] = [];
    while (size) {
      const node = stack.shift();
      if (node) {
        record.push(node.val);
        node.left && stack.push(node.left);
        node.right && stack.push(node.right);
      }

      size--;
    }

    record.length && res.push(record);
  }

  return res;
}

File: docs/algorithms/code/algorithms/levelOrder.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func levelOrder(root *TreeNode) [][]int {
  if root == nil {
    return [][]int{};
  } 

  res := [][]int{};
  queue := []*TreeNode{root};

	for len(queue) > 0 {
		size := len(queue);
		level := []int{};

		for i:=0; i < size; i++ {
			node := queue[0];
			queue = queue[1:];
			level = append(level, node.Val);

			if node.Left != nil {
				queue = append(queue, node.Left);
			}

			if node.Right != nil {
				queue = append(queue, node.Right);
			}
		}

    res = append(res, level)
	}

	return res;
}